server/build.go

```diff
@@ -111,6 +112,23 @@ func (task *BuildTask) Build() (esm *ESM, err error) {
 		hasher.Write([]byte(task.ID()))
 		task.wd = path.Join(os.TempDir(), fmt.Sprintf("esm-build-%s-%s", hex.EncodeToString(hasher.Sum(nil)), rs.Hex.String(8)))
 		ensureDir(task.wd)
+
+		rcFilePath := path.Join(task.wd, ".npmrc")
+
+		if !fileExists(rcFilePath) {
+			err = ioutil.WriteFile(
+				rcFilePath,
+				[]byte("_authToken=${ESM_NPM_TOKEN}"),
```

Comment:
does the npmrc supports `${ESM_NPM_TOKEN}` expression?

Reply:
Yup, the `.npmrc` is supported, Is designed for [private packages in CI/CD](https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow#create-and-check-in-a-project-specific-npmrc-file).
But `.yarnrc` is not, and but `.yarnrc.yml` is supported 🤣 Yarn is so confused.

Actually, I tried to use other ways to implement it,  such as:
- Environment variables begin with `npm_config_`, but yarn will lower case all suffix, so the `_authToken` to be transformed to `_authtoken`, the `_auth_token` to be transformed to `_auth-token` 
- Use `--use-yarnrc <path>` for providing a specific `.yarnrc` file, but it was not supported to decode the environment variable expression

Reply:
thanks for the explain! i didn't known this before